src/Traits/CanRedeemVouchers.php

```diff
@@ -109,17 +109,20 @@ public function redeemVoucher(string $voucher): Voucher
      * Check whether the user instance can redeem a voucher or voucher code.
      *
      * @param Voucher|string $voucher
-     * @param \Illuminate\Database\Eloquent\Model|string|null $product
+     * @param Model|array|string|null $product
      * @return bool
      */
-    public function canRedeem(Voucher|string $voucher, Model|string|null $product = null): bool
+    public function canRedeem(Voucher|string $voucher, Model|array|string|null $product = null): bool
     {
         if (is_string($voucher)) {
             $voucher = Vouchers::check($voucher);
         }
 
-        if ($product && !$voucher->isItem($product)) {
-            return false;
+        if ($product !== null) {
+            if ($product instanceof Model || is_string($product)) {
+                $product = [$product];
+            }
+            return $voucher->isAnyItem($product);
```

Comment:
Better to use `isAllowed` here? This would ensure ALL items can redeem the voucher.